src/cdomains/apron/apronDomain.apron.ml

```diff
@@ -834,12 +834,14 @@ struct
   let widen x y =
     let x_env = A.env x in
     let y_env = A.env y in
-    if Environment.equal x_env y_env then
-      (* widen if env didn't increase *)
-      if GobConfig.get_bool "ana.apron.threshold_widening" then
+    if Environment.equal x_env y_env  then
+      if GobConfig.get_bool "ana.apron.threshold_widening" && Oct.manager_is_oct Man.mgr then
+        let octmgr = Oct.manager_to_oct Man.mgr in
         let ts = Lazy.force widening_thresholds_apron in
-        let r = Oct.widening_thresholds Man.mgr (Abstract1.abstract0 x) (Abstract1.abstract0 y) ts in
-        {x with abstract0 = r}
+        let x_oct = Oct.Abstract1.to_oct x in
+        let y_oct = Oct.Abstract1.to_oct y in
+        let r = Oct.widening_thresholds octmgr (Abstract1.abstract0 x_oct) (Abstract1.abstract0 y_oct) ts in
+        Oct.Abstract1.of_oct {x_oct with abstract0 = r}
```

Comment:
Instead of doing these (literally magic) conversions, it might be neater to just add a `widen_threshold` function to our own `Manager` signature, such that octagons can specially implement it in a safe way, while the others can just do normal widening for the time being.